src/lib.mli

```diff
@@ -227,6 +237,7 @@ module DB : sig
   val create
     :  ?parent:t
     -> resolve:(Lib_name.t -> Resolve_result.t)
+    -> find_implementations:(Lib_name.t -> Lib_info.t list Variant.Map.t)
```

Comment:
Could you comment on the necessity of this new field? Why can't we just use `all` and iterate over all the public libraries this way for example? Is it because we want to avoid resolving them?

Reply:
The goal was to avoid resolving them. Before coming to this solution I wanted to put possible implementations in the Lib.t type but as you mentioned after this introduces a dependency cycle on instantiation.

Reply:
Yes, indeed that your first attempt would have been ideal. Unfortunately, we will need forward references before it's possible.

Why is resolving the implementations a problem however? Are there situations where we just need the Lib_info.t? We can't really do much with this value apart from resolving it into a proper library.

Reply:
It's not really a problem, I just tried to do a minimal amount of computation when evaluating variants: we only need to resolve the implementations that are actually selected.